rwengine/include/script/ScriptTypes.hpp

```diff
@@ -27,6 +29,145 @@ typedef uint16_t SCMOpcode;
 typedef char SCMByte;
 typedef unsigned int SCMAddress;
 
+///////////////////////////////////////////////////////////
+// Script Implementation Types
+
+using ScriptInt = int32_t;
+using ScriptLabel = int32_t;
+using ScriptIntGlobal = int32_t&;
+using ScriptIntLocal = int32_t&;
+using ScriptFloat = float;
+using ScriptFloatGlobal = float&;
+using ScriptFloatLocal = float&;
+using ScriptBoolean = int32_t;
+
+using ScriptVec2 = glm::vec2;
+using ScriptVec3 = glm::vec3;
+
+using ScriptRGB = glm::u8vec3;
+using ScriptRGBA = glm::u8vec4;
+
+using ScriptModelID = int32_t;
+using ScriptString = char[8];
+
+
+/**
+ * Helper class for transparently handling integer handles for objects
+ */
+template <class T>
+struct ScriptObjectType {
+	/**
+	 * @brief m_id VM Memory containing the object handler
+	 */
+	ScriptInt* m_id;
+	/**
+	 * @brief m_object Real object instance
+	 */
+	T* m_object;
+
+	ScriptObjectType(ScriptInt* var, GameObject* object)
+		: m_id(var)
+		, m_object(static_cast<T*>(object))
+	{
+	}
+
+	ScriptObjectType(ScriptInt* var, T* object)
+		: m_id(var)
+		, m_object(object)
+	{
+	}
+
+	/**
+	 * Assigns the memory location to the ID of the given instance
+	 */
+	T* operator = (T* object)
+	{
+		RW_CHECK(m_id != nullptr, "ScriptObjectType has pointer to null memory location");
+		*m_id = object->getScriptID();
+		return object;
+	}
+
+	T* operator -> () const
+	{
+		RW_CHECK(m_object != nullptr, "Dereferencing ScriptObjectType with null instance");
+		return m_object;
+	}
+};
+
+using ScriptObject = ScriptObjectType<InstanceObject>;
+using ScriptPlayer = ScriptObjectType<PlayerController>;
+using ScriptVehicle = ScriptObjectType<VehicleObject>;
+using ScriptCharacter = ScriptObjectType<CharacterObject>;
+using ScriptPickup = ScriptObjectType<PickupObject>;
+
+struct VehicleGenerator;
+struct BlipData;
+struct GarageInfo;
+
+using ScriptVehicleGenerator = ScriptObjectType<VehicleGenerator>;
+using ScriptBlip = ScriptObjectType<BlipData>;
+using ScriptGarage = ScriptObjectType<GarageInfo>;
+
+/* Not Implemented Yet */
+using ScriptSound = ScriptObjectType<int>;
```

Comment:
They're not implemented yet, they'll be replaced with a proper type once the functionality for those are implemented.